### PR TITLE
Json structure streaming module place holder validation

### PR DESCRIPTION
**Pull request by YooSunYoung**
These validation steps will be used in the integration test as well.

### CODE REVIEW

File: src/beamlime/applications/_nexus_helpers.py [hunk 1]
```diff
@@ -91,29 +91,82 @@ class RunStartInfo:
...
+    """Validate the module keys."""
+    key_set = {key for key, _ in key_value_pairs}
+    key_list = [key for key, _ in key_value_pairs]
+    for unique_key in key_set:
+        key_list.remove(unique_key)
+    # ``key_list`` should be empty if all keys are unique.
+    if len(key_list) != 0:
+        raise InvalidNexusStructureError(
+            "Duplicate module place holder(s) found in the nexus structure: ", key_list
+        )
```
Comment (MridulS) [inline src/beamlime/applications/_nexus_helpers.py:126]:
```suggestion
    from collections import Counter
    key_list = [key for key, _ in key_value_pairs]
    key_counts = Counter(key_list)
    if max(key_counts.values()) > 1:
        raise InvalidNexusStructureError(
            "Duplicate module place holder(s) found in the nexus structure: ", key_list
        )
```
Just a styling choice, feel free to ignore :)

Comment (YooSunYoung) [reply src/beamlime/applications/_nexus_helpers.py:126]: Oh I just didn't know of this. I'll slightly update it with filtering the counts. 
The idea was that ``key_list`` only contains the duplicating keys for the error message.
Thanks.

File: src/beamlime/applications/_nexus_helpers.py [hunk 2]
```diff
@@ -91,29 +91,82 @@ class RunStartInfo:
...
+
+    Raises
+    ------
+    ValueError
+        If the structure does not have a valid module place holder configuration.
+        - Contains ``topic`` and ``source``.
+        - ``module`` - ``topic`` - ``source`` combination is unique.
+
+    """
+    # Collect stream module key-value pairs as tuple
```
[the remark this answers is not in the document]
Comment (YooSunYoung) [reply src/beamlime/applications/_nexus_helpers.py:140]: logging will be handled by the handlers and daemons.
I think it's a good idea to log these collected modules.

[the remark this answers is not in the document]
Comment (YooSunYoung) [reply src/beamlime/applications/_nexus_helpers.py:142]: @MridulS Done!

**Review by MridulS — APPROVED** [review]

**Pull request merged**